wordpress-mobile/WordPress-iOS
# Gutenberg: Update the tracker for unsupported blocks

This PR updates the interface implementation of the method that is used to keep track of unsupported blocks in the current post/page.

Prior to this PR, the JS side was sending a simple flag to signal the presence of unsupported blocks in the post. With the improved version of the JS side tracker, it will send the full list of unsupported block names. For now we're just checking if the list is empty or not, and then bump the same Analytics has_unsupported_blocks as before.

Ref:
WPAndroid PR: https://github.com/wordpress-mobile/WordPress-Android/pull/10170
Gutenberg Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/1199
Gutenberg PR: https://github.com/WordPress/gutenberg/pull/16434

To test:
- Ensure that you have the block editor switch ON on Me -> App Settings
- Create a new post with a couple of paragraph and save it (as draft or publish)
- Open the saved post again
  - Check that the `editor_session_start` track event is registered with `has_unsupported_blocks: 0` property:

```
🔵 Tracked: editor_session_start <blog_type: wpcom, content_type: gutenberg, editor: gutenberg, has_unsupported_blocks: 0, post_type: post, session_id: A5877C9D-646A-4DBC-93B9-23778CAFDA84>
```
- Open the same post on web and add a few unsupported block (i.e: Gallery) and save.
- Refresh the post lists on mobile and open the edited post.
  - Check that the `editor_session_start` track event is registered with `has_unsupported_blocks: 1` property:
```
🔵 Tracked: editor_session_start <blog_type: wpcom, content_type: gutenberg, editor: gutenberg, has_unsupported_blocks: 1, post_type: post, session_id: 47A54BF2-A70F-466D-A76F-BBD32545AF71>
```

Note that this is a Bool value, so even if there are many unsupported blocks, `has_unsupported_blocks: 1` is expected.
